src/types.rs

```diff
@@ -50,26 +50,30 @@ pub type R64 = NoisyFloat<f64, FiniteChecker>;
 
 /// Shorthand for `N32::new(value)`.
 #[inline]
-pub fn n32(value: f32) -> N32 {
-    N32::new(value)
+pub const fn n32(value: f32) -> N32 {
+    assert!(!value.is_nan());
```

Comment:
I like the idea, now that asserts are stable in const functions. However, these need to be `debug_assert` statements rather than `assert` statements. The functions will not always be used in a const context, so we need to be sure not to incur a runtime overhead in production.

Reply:
Thanks for the feedback! I overlooked that this crate in general only uses debug assertions. I'll change that.